mirascope/core/base/_utils/parse_content_template.py

```diff
@@ -0,0 +1,115 @@
+"""This module provides a function to parse content parts from a prompt template."""
+
+import imghdr
```

Comment:
looks like this is getting deprecated in python 3.13. either implement the alternative or put a note https://docs.python.org/3/library/imghdr.html

Reply:
Done. I wrote our own detection function and also added handling for each provider to throw an error if the user provides an image type that provider doesn't support.